### PR TITLE
Use RWMutex in some simple cases

### PR DESCRIPTION
**Pull request by MaruHyl**
#435 I need to reopen a PR because of force-pushed.

### CODE REVIEW

File: nats.go [hunk 1]
```diff
@@ -2834,9 +2840,10 @@ func (nc *Conn) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg) (*Sub
 
 // NumSubscriptions returns active number of subscriptions.
 func (nc *Conn) NumSubscriptions() int {
-	nc.mu.Lock()
-	defer nc.mu.Unlock()
-	return len(nc.subs)
+	nc.mu.RLock()
```
Comment (derekcollison) [inline nats.go]: Prefer the defer version, and all of these are not critical perf wise, so let's revert these.

**Comment by MaruHyl** [comment]
@derekcollison okay, reverted.

File: nats.go (hunk 1 above)
Comment (MaruHyl) [reply nats.go]: done.

**Comment by MaruHyl** [comment]
@derekcollison  Travis done.

**Pull request merged**